packages/kbn-cell-actions/src/actions/utils.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { KBN_FIELD_TYPES } from '@kbn/field-types';
+import { isBoolean, isNumber, isString } from 'lodash/fp';
+import { Serializable } from '@kbn/utility-types';
+import { DefaultActionsPrimitiveValue } from './types';
+
+export const SUPPORTED_KBN_TYPES = [
+  KBN_FIELD_TYPES.DATE,
+  KBN_FIELD_TYPES.IP,
+  KBN_FIELD_TYPES.STRING,
+  KBN_FIELD_TYPES.NUMBER,
+  KBN_FIELD_TYPES.BOOLEAN,
+];
+
+export const isTypeSupportedByDefaultActions = (kbnFieldType: KBN_FIELD_TYPES) =>
+  SUPPORTED_KBN_TYPES.includes(kbnFieldType);
+
+export const isNonNullablePrimitiveValue = (
+  value: Serializable
+): value is DefaultActionsPrimitiveValue => isString(value) || isNumber(value) || isBoolean(value);
+
+export const isPrimitiveValue = (value: Serializable): value is DefaultActionsPrimitiveValue =>
+  value == null || isNonNullablePrimitiveValue(value);
+
+/**
+ * Unfortunately, we can't create type guards for this function because `SerializableArray`
+ * is an interface that extends Array, which is incompatible with `Array<number | boolean | string>`.
+ *
+ * Whenever this function returns true, the value is guaranteed to be a `DefaultActionsSupportedValue`.
+ */
+export const isValueSupportedByDefaultActions = (value: Serializable) =>
+  isPrimitiveValue(value) || (Array.isArray(value) && value.every(isNonNullablePrimitiveValue));
```

Comment:
I am still not convinced we should drop support for `[null | undefined]`. 
I remember we had to take it into account in some actions because we were receiving it in the alerts table.
I am talking about an array with only one `null` value:
```suggestion
export const isValueSupportedByDefaultActions = (value: Serializable) =>
  isPrimitiveValue(value) || (Array.isArray(value) && (value.length < 2 || value.every(isNonNullablePrimitiveValue)));
```

Reply:
What about empty arrays? maybe we should also return `false`? 🤔

Reply:
And with `[1, 'hello', false]`? Should we accept mixed values? `Serializable` allows it, but if we receive an array in the `field.value` all values in the array should be consistent with the `field.type`? 
Probably it is worth checking it, not especially for our existing use cases, but for others that may come. wdyt?

Reply:
Let me address all comments for the default actions inside the CellActions package:

<hr />

### 1. Support for `[null | undefined]` (nullable arrays)


Filter-in and Filter-out actions throw a runtime error when called with the value `[null]`. So, if the alerts table calls cell actions with `[null]`, we have a bug.
<img width="581" alt="Screenshot 2023-06-28 at 10 33 29" src="https://github.com/elastic/kibana/assets/1490444/a8e96a5e-54db-486c-a3a0-d16c2bb2a906">

`isEmptyFilterValue`  also returns false for an array of nullable values:

```ts
export const isEmptyFilterValue = (
  value: DefaultActionsSupportedValue
): value is null | undefined | never[] =>
  value == null || value === '' || (isArray(value) && value.length === 0); 
```

To allow `[null]`, we first must update the actions to support it. What would be the expected behaviour? An `exists` filter?

<hr />


### 2. Support for `[]` (empty arrays)

Default cell actions currently support empty arrays. `copy_to_clipboard` copies the field name, and `filter_in` and `filter_out` build an `exists` filter. You can find the check inside `isEmptyFilterValue`. So, I don't see a good reason for dropping the support of empty arrays. They are an expected value for aggregations.

<hr />

### 3. Support for `[1, 'hello', false]` (mixed-typed array)

This is an interesting point. The default actions work with mixed-typed arrays, but I can't imagine a valid use case. Could a field have multiple types? And it would be very inconsistent because we only support one KBN_TYPE. 🤔 
I am inclined to be cautious and restrict this scenario. We could easily add it back if we find a good use case. 


### Extra
While testing some mentioned scenarios, I noticed that we currently support `['hello', null]` (mixed-nullable-typed array), but it also throws an error. I will restrict it when implementing number 3.

Reply:
## 1
> To allow `[null]`, we first must update the actions to support it. What would be the expected behavior? An exists filter?

Yes, maybe I introduced the bug when I did the latest changes to `createFilter` 🙃.

## 2
> Default cell actions currently support empty arrays

Sounds good, it is consistent.

## 3
> I am inclined to be cautious and restrict this scenario

Agree. Maybe the only valid case would be a combination of valid types and _nullables_ e.g. `[null, 'valid value']`. Not sure it will ever happen (maybe coming from an aggregation?), but I would not consider that "invalid" (it's like it has two values one of which is missing), probably we could process it like `['valid value']`?

### Suggestion

Having all this in mind, in the default actions, when the value is an array, maybe it would be a good idea to reduce the values array and remove all `null` | `undefined`, before checking all values in the array are the same type.
Cases like `[null, 'valid value']` would be converted to `['valid value']`, and if everything is _nullable_ it will be converted to `[]`, so we will apply the `not exists` correctly.
wtyt?